cython/gtsam/utils/logging_optimizer.py

```diff
@@ -46,5 +46,7 @@ def gtsam_optimize(optimizer,
     def check_convergence(optimizer, current_error, new_error):
         return (optimizer.iterations() >= params.getMaxIterations()) or (
             gtsam.checkConvergence(params.getRelativeErrorTol(), params.getAbsoluteErrorTol(), params.getErrorTol(),
-                                   current_error, new_error))
+                                   current_error, new_error)) or (
+            type(optimizer).__name__ == "LevenbergMarquardtOptimizer" and optimizer.lambda_() > params.getlambdaUpperBound())
```

Comment:
Oops, did you actually run the test? it should be lambda as that is wrapped. No problem with reserved word as it's a method I think.

Reply:
I actually run the test, and indeed lambda is not usable, and is converted as `lambda_` in the wrapper.

Reply:
The *cython* wrapper?

Reply:
Yes

Reply:
https://colab.research.google.com/drive/1AJ-6mlrDhhV3Uqbn7ahHDEy6dUWsmsDe?usp=sharing

Reply:
But how did it ever work for me then?

Reply:
The logic is in `wrap/FullyOverloadedFunction.cpp`:
```
/// Cython: Rename functions which names are python keywords
  static const std::array<std::string, 2> pythonKeywords;
  static std::string pyRename(const std::string& name) {
    if (std::find(pythonKeywords.begin(), pythonKeywords.end(), name) ==
        pythonKeywords.end())
      return name;
    else
      return name + "_";
  }
```